remoulade/state/backends/postgres.py

```diff
@@ -0,0 +1,114 @@
+    actor_name = Column(String)
+    args = Column(JSONB)
+    kwargs = Column(JSONB)
+    options = Column(JSONB)
+    priority = Column(SmallInteger)
+    enqueued_datetime = Column(DateTime)
+    started_datetime = Column(DateTime)
+    end_datetime = Column(DateTime)
+    group_id = Column(String)
+
+    def as_state(self) -> State:
+        state_dict = {}
+        mapper = inspect(StoredState)
+        for column in mapper.attrs:
+            if getattr(self, column.key) is not None:
+                if column.key in ["enqueued_datetime", "started_datetime", "end_datetime"]:
+                    state_dict[column.key] = utc.localize(getattr(self, column.key)).isoformat()
+                else:
+                    state_dict[column.key] = getattr(self, column.key)
+        return State.from_dict(state_dict)
+
+
+class PostgresBackend(StateBackend):
+    def __init__(self, *, namespace="remoulade-state", encoder=None, client=None, url=None):
+        super().__init__(namespace=namespace, encoder=encoder)
+        db_string = "postgresql://wiremind_owner@localhost:5432/wiremind"
+        db = create_engine(db_string)
+        sm = sessionmaker(db)
+        self.session = sm()
+
+    def get_state(self, message_id):
+        state = self.session.query(StoredState).filter_by(message_id=message_id).first()
+        if not state:
+            return None
+        return state.as_state()
+
+    def set_state(self, state, ttl=3600):
+        stored_state = self.session.query(StoredState).filter_by(message_id=state.message_id).first()
+        if stored_state is None:
+            stored_state = StoredState(state)
+            self.session.add(stored_state)
+        else:
+            for key, value in state.as_dict().items():
+                setattr(stored_state, key, value)
+
+        self.session.commit()
+
+    def get_states(
+        self,
+        *,
+        size: int = 50,
+        offset: int = 0,
+        selected_actors: List[str] = None,
+        selected_statuses: List[str] = None,
+        selected_ids: List[str] = None,
+        start_datetime: datetime.datetime = None,
+        end_datetime: datetime.datetime = None,
+        sort_column: str = None,
+        sort_direction: str = None,
+        **kwargs,
+    ):
+        data = self.session.query(StoredState)
+        if selected_actors is not None:
+            data = data.filter(StoredState.actor_name.in_(selected_actors))
+        if selected_statuses is not None:
+            data = data.filter(StoredState.status.in_(selected_statuses))
+        if selected_ids is not None:
+            data = data.filter(StoredState.message_id.in_(selected_ids))
+        if start_datetime is not None:
+            data = data.filter(StoredState.enqueued_datetime > start_datetime)
+        if end_datetime is not None:
+            data = data.filter(StoredState.enqueued_datetime < end_datetime)
+        if sort_column in State._fields and sort_direction in ["asc", "desc"]:
```

Comment:
I think you can remove ` and sort_direction in ["asc", "desc"]`, and do `query = query.order_by(f"{sort_column} {sort_direction}"))`